lib/pbench/test/unit/server/test_datasets_daterange.py

```diff
@@ -43,15 +44,11 @@ def query_api(
 
         return query_api
 
-    def token(self, client, config: PbenchServerConfig, user: str) -> str:
-        response = client.post(
-            f"{config.rest_uri}/login",
-            json={"username": user, "password": "12345"},
-        )
-        assert response.status_code == HTTPStatus.OK
-        data = response.json
-        assert data["auth_token"]
-        return data["auth_token"]
+    def token(self, user: str) -> str:
+        roles = None
+        if user == "test_admin":
+            roles = ["ADMIN"]
+        return generate_token(username=user, pbench_client_roles=roles)
```

Comment:
You're repeating this; it should be factored out into a higher level, possibly in `pbench.test.unit.server` or in `conftest.py` along with `generate_token`.

Reply:
And, that looks like a nice place for a ternary....  😉

Reply:
Yes, ternary would be better here. (Webb is so fond of ternary syntax I must say :stuck_out_tongue_closed_eyes:)

Reply:
Ternary is great when `black` doesn't char it to unreadability, and sadly in most of these instances that hope has been dashed...